python/lsst/obs/base/formatters/fitsExposure.py

```diff
@@ -22,27 +22,149 @@
 __all__ = ("FitsExposureFormatter", "FitsImageFormatter", "FitsMaskFormatter",
            "FitsMaskedImageFormatter")
 
+from abc import abstractmethod
 import warnings
 
-from astro_metadata_translator import fix_header
 from lsst.daf.base import PropertySet
 from lsst.daf.butler import Formatter
-# Do not use ExposureFitsReader.readMetadata because that strips
-# out lots of headers and there is no way to recover them
-from lsst.afw.fits import readMetadata
+from lsst.daf.butler.core.utils import cached_getter
 from lsst.afw.image import ExposureFitsReader, ImageFitsReader, MaskFitsReader, MaskedImageFitsReader
 from lsst.afw.image import ExposureInfo, FilterLabel
 # Needed for ApCorrMap to resolve properly
 from lsst.afw.math import BoundedField  # noqa: F401
 
 
 class FitsImageFormatterBase(Formatter):
-    """Base class interface for reading and writing afw images to and from
-    FITS files.
+    """Base class formatter for image-like storage classes stored via FITS.
 
-    This Formatter supports write recipes.
+    Notes
+    -----
+    This class makes no assumptions about how many HDUs are used to represent
+    the image on disk, and includes no support for writing.  It's really just a
+    collection of miscellaneous boilerplate common to all FITS image
+    formatters.
 
-    Each ``FitsImageFormatterBase`` recipe for FITS compression should
+    Concrete subclasses must implement `readComponent`, `readFull`, and `write`
+    (even if just to disable them by raising an exception).
+    """
+
+    extension = ".fits"
+    supportedExtensions = frozenset({".fits", ".fits.gz", ".fits.fz", ".fz", ".fit"})
+
+    unsupportedParameters = {}
+    """Support all parameters."""
+
+    @property
+    @cached_getter
+    def checked_parameters(self):
+        """The parameters passed by the butler user, after checking them
+        against the storage class and transforming `None` into an empty `dict`
+        (`dict`).
+
+        This is computed on first use and then cached.  It should never be
+        accessed when writing.  Subclasses that need additional checking should
+        delegate to `super` and then check the result before returning it.
+        """
+        parameters = self.fileDescriptor.parameters
+        if parameters is None:
+            parameters = {}
+        self.fileDescriptor.storageClass.validateParameters(parameters)
```

Comment:
Is this something the base `Formatter` or `FileDatastore` (and/or maybe assemblers) could guarantee, post-DM-29964, so concrete formatters don't have to do it themselves?